x-pack/plugins/lens/public/editor_frame_service/error_helper.tsx

```diff
@@ -101,13 +99,15 @@ function getNestedErrorClause(e: ErrorCause | Reason): ReasonDescription[] {
 
 function getErrorSources(e: Error) {
   if (isRequestError(e)) {
-    return getNestedErrorClause(e.body!.attributes!.error as ErrorCause);
+    return getNestedErrorClause(e.body!.attributes!.error as estypes.ErrorCause);
   }
   if (isEsError(e)) {
-    if (e.attributes?.reason) {
-      return getNestedErrorClause(e.attributes);
+    if (e.attributes?.error?.reason) {
+      return getNestedErrorClause(e.attributes.error);
+    }
+    if (e.attributes?.error?.caused_by) {
```

Comment:
Not sure I understand why this guard was inserted. It doesn't look like we gated the logic by the existence of `caused_by` before.

Reply:
attributes.error is typed as optional.

Reply:
It looks like the previous logic assumed that `caused_by` was defined (type cast) which it must always have been since `getNestedErrorClause` would have errored if passed `undefined`.

Given this, I _think_ changing from casting to checking is okay because it should not change the amount of times this branch gets entered.